py/datastore_service.py

```diff
@@ -103,11 +122,27 @@ def handle_request(self, post):
     #get the reporter data
     reports = self.parse_reports(post)
 
-    #TODO: build a query
-    #for report in reports:
+    try:   
+      # get the provider. 
+      provider = reports['provider']
+
+      # get the reports and loop over to get the rest of the data.
+      for report in reports['reports']:
+        segment_id = report['segment_id']
+        prev_segment_id = report.get('prev_segment_id', None)
+        mode = report.get('mode', 'NULL')
+        start_time = report['start_time']
+        end_time = report['end_time']
+        length = report['length']
+
+        # send it to the cursor.
+        self.server.sql_conn.cursor().execute("execute report (%s,%s,%s,%s,%s,%s,%s)",
```

Comment:
...this can be `self.server.cursor.execute(...` so that you dont have to get a cursor every time